### ISSUE TITLE
Imply keyword throws exception when performing QUANTIFIERS_REMOVING

### ISSUE THREAD

**Issue by MFaisalZaki** [aiplan4eu/unified-planning#593]
In one of the IPC-2006 domains (e.g. trucks), the domain model has `forall` and `imply` keywords in some actions' preconditions. 
I tried to apply a QUANTIFIERS_REMOVING followed by GROUNDING, but I do get this error:

![Screenshot from 2024-04-11 13-03-41](https://github.com/aiplan4eu/unified-planning/assets/9129265/5d03c20d-adf5-4de5-a217-74944f2d8237)

Here is the code to reproduce:
```
from unified_planning.shortcuts import *
from unified_planning.engines import CompilationKind
from unified_planning.io import PDDLReader

task = PDDLReader().parse_problem('./domain.pddl', './problem.pddl')


compilation_kinds_to_apply = [
    CompilationKind.QUANTIFIERS_REMOVING,
    CompilationKind.CONDITIONAL_EFFECTS_REMOVING,
    CompilationKind.GROUNDING
]

compiled_results = None
for compilation_kind in compilation_kinds_to_apply:
    with Compiler(problem_kind=task.kind, compilation_kind=compilation_kind) as compiler:
        task = compiler.compile(task, compilation_kind)
        task = task.problem
```
and here is the used [planning task](https://github.com/aiplan4eu/unified-planning/files/14945820/trucks-2006.zip)

**Comment by MFaisalZaki** [comment]
Thanks a lot @Framba-Luca.

**Comment by alvalentini** [comment]
I close the issue since it is solved in #594
